base/compiler/typelattice.jl

```diff
@@ -84,8 +84,10 @@ struct NotFound end
 
 const NOT_FOUND = NotFound()
 
-const CompilerTypes = Union{Type, MaybeUndef, Const, Conditional, NotFound, PartialStruct}
-==(x::CompilerTypes, y::CompilerTypes) = x === y
+const CompilerTypes = Union{MaybeUndef, Const, Conditional, NotFound, PartialStruct}
+==(x::CompilerTypes, y::CompilerTypes) = false
```

Comment:
I guess we never use this method, but this definition still seems a little hazardous?